### PR TITLE
Compressedrefs - Map memory below 4GB - OSX

### PR DESCRIPTION
**Pull request by babsingh**
**1) Allow memory to be mapped below 4GB on OSX**

  On OSX 64-bit, linker sets page zero size to 4GB. This prevents memory
  to be mapped below 4GB.

  On OSX 32-bit, page zero size is set to 4KB by default.

  This change will set page zero size to 4KB on 64-bit architectures. This
  will allow memory to be mapped below 4GB.

**2) Enable support for compressedrefs feature in OMR**

  For compressedrefs, memory needs to be mapped below 4GB.

  Currently, code for allocating/mapping memory below 4GB is disabled on
  OSX.

  Specifying -pagezero_size <SIZE> while creating main executables will
  allow us to map memory below 4GB.

  Thus, re-enabling code for allocating/mapping memory below 4GB:
  omrmem_allocate_memory32 and supporting test code.

Signed-off-by: Babneet Singh <sbabneet@ca.ibm.com>

### CODE REVIEW

**Comment by babsingh** [comment]
fyi - @DanHeidinga @charliegracie @youngar @dnakamura

File: cmake/modules/platform/os/osx.cmake [hunk 1]
```diff
@@ -28,3 +28,10 @@ list(APPEND OMR_PLATFORM_DEFINITIONS
 # of the OMR code should be heavily reduced. In the mean time, we keep
 # the distinction
 list(APPEND TR_COMPILE_DEFINITIONS -DSUPPORTS_THREAD_LOCAL -DOSX)
+
+if(OMR_ENV_DATA64)
```
Comment (charliegracie) [inline cmake/modules/platform/os/osx.cmake]: this should be if 64 bit and compressed references I think

**Comment by charliegracie** [comment]
The commit message for `Enabled compressedrefs feature in OMR` is a little misleading. That commit is not actually enabling compressed references in OMR or OMR testing but rather allowing the feature to be enabled.

**Comment by charliegracie** [comment]
@genie-omr build all

**Comment by DanHeidinga** [comment]
related to eclipse/openj9#36

**Pull request merged**